blockchain/chain/src/store/chain_store.rs

```diff
@@ -560,12 +563,15 @@ where
         Self::walk_snapshot(tipset, recent_roots, |cid| {
             let tx_clone = tx.clone();
             async move {
+                let prefix = Code::try_from(Prefix::from(&cid).mh_type)?;
```

Comment:
I don't particularly like this `Prefix` struct that ultimately got [removed](https://github.com/multiformats/rust-cid/pull/49) from the `rust-cid`.

I do believe this should work and we could then get rid of the `blockchain/chain/src/store/prefix.rs`

Could you please try it out?
```suggestion
                let prefix = Code::try_from(cid.hash().code())?;
```

Reply:
Perhaps even better, let me tinker around.

Reply:
Sounds good, IIRC I did try `cid.hash().code()` but that gave me an error during snapshot export.

Reply:
@jdjaustin try to cherry-pick this commit https://github.com/ChainSafe/forest/commit/c131ab21a7b80b39088bc62089c15abdf54b66c0

It didn't fail on export and imported the snapshot fine, but I haven't checked that the checksums still match. Hopefully, they still do. Could you please verify this?

Reply:
Yes they still match. Compiling Forest from that branch, here are the results for the calibnet exports:
![Screenshot from 2023-02-16 12-27-51](https://user-images.githubusercontent.com/11778588/219455638-529fa63a-9be3-46ef-a35b-e65cb62a7091.png)
<img width="563" alt="Screen Shot 2023-02-16 at 12 28 11 PM" src="https://user-images.githubusercontent.com/11778588/219455665-519b2179-d72c-471b-9272-6477908706be.png">